src/plugins/image/editing/image-editor.js

```diff
@@ -69,24 +69,67 @@ class ImageEditor extends React.Component {
     this.changeContent({ text: newValue });
   }
 
-  handleHoverCopyrightInfoValueChanged(event) {
-    const newValue = event.target.value;
-    const hover = { ...this.props.content.hover };
-    hover.text = newValue;
-    this.changeContent({ hover });
+  handleOrientationValueChanged(e) {
+    const effect = {
+      ...this.props.content.effect, orientation: e.target.value
+    };
+
+    this.changeContent({ effect });
   }
 
-  handleHoverSwitchChange(checked) {
-    if (checked) {
-      this.changeContent({ hover: { type: 'internal', url: '', text: '' } });
-    } else {
-      this.changeContent({ hover: null });
+  handleEffectOptionChange(e) {
+    const baseProps = {
+      sourceType: SOURCE_TYPE.internal,
+      sourceUrl: '',
+      text: ''
+    };
+
+    switch (e.target.value) {
+      case EFFECT_TYPE.reveal:
+        this.changeContent({
+          effect: {
+            ...baseProps,
+            type: EFFECT_TYPE.reveal,
+            startPosition: 10,
+            orientation: ORIENTATION.horizontal
+          }
+        });
+        break;
+      case EFFECT_TYPE.hover:
+        this.changeContent({
+          effect: {
+            ...baseProps,
+            type: EFFECT_TYPE.hover
+          }
+        });
+        break;
+      case EFFECT_TYPE.none:
+        this.changeContent({ effect: null });
+        break;
+      default:
+        this.changeContent({ effect: null });
+
     }
   }
 
+  handleEffectCopyrightInfoValueChanged(event) {
+    const newValue = event.target.value;
+    const effect = { ...this.props.content.effect };
+    effect.text = newValue;
+    this.changeContent({ effect });
+  }
+
+  handleStartPositionValueChanged(newPosition) {
+    const effect = { ...this.props.content.effect };
+    effect.startPosition = Math.max(0, Math.min(100, newPosition));
+    this.changeContent({ effect });
+  }
+
   render() {
     const { docKey, content, clientConfig, t } = this.props;
-    const { type, url, maxWidth, text, hover } = content;
+    const { sourceType, sourceUrl, maxWidth, text, effect } = content;
+
+    const effectType = (effect && effect.type) || EFFECT_TYPE.none;
```

Comment:
could be: `const effectType = effect?.type || EFFECT_TYPE.none;` (what do you think about this coding style?)

Reply:
I thought it comes from typescript and I was dying inside to write this...

Reply:
:smile: